packages/edit-site/src/components/page-pages/index.js

```diff
@@ -22,6 +22,12 @@ import Media from '../media';
 
 const EMPTY_ARRAY = [];
 const EMPTY_OBJECT = {};
+const defaultConfigPerViewType = {
```

Comment:
Each time a user switches to a given view type (layout), this object is added to the config of the view automatically under "view.layoutConfig". I guess in my mind `layoutConfig` is a property that serves to hold "view type specific config".

One downside is that switching back and forth resets this config.

Reply:
I don't really get why we need a layout config at this point. Why these are not part of the view? Should we update the API to pass multiple views?

Reply:
It it part of the view, but I decided to put it under a "subkey" that way we know that these configs are "layout specific", which allows us to reset these when you switch "layouts" and not write any specific code for it.

For instance when you switch from table to grid, you need to provide the "mediaField" and when you switch from grid to table, you need to remove the "mediaField". By using a sub key to hold all of these layout specific configs, we're able to do this automatically as soon as the view.type changes. So the layout switcher UI component doesn't need to know which view types are available and what are their specific configs.

Reply:
Okay, still I think it's probably the time to pass multiple views and not keep one and update part of it. That's the end goal too, as we can have multiple views of the same type(two grid views with different image as the primary one, etc..). Part of the API is the cross view config, like `search` etc.., and they should just be updated when changing `view type`(list to grid etc..).

Makes sense what I'm suggesting here?

Reply:
I'm not following, a view can only have one type and a config and the user should be able to change the view type of its view (or saved view), so we need to handle these switches anyway.

Reply:
I don't disagree with that. In essence I'm saying that I think it's better to keep track of an array of views(with an `isActive` prop or something) and when a user changes a view, switch to a whole new view object and not update a single view like we do now. In my mind this seems 'simpler' let's say, than trying to wrangle merges, as new props are added to each view config.

Reply:
@ntsekouras That sounds a bit weird to me, if you have filtering, sorting, visible fields in place, I'm thinking we would want to keep them. But yes, it's a good question to ask, what happens to the current view config when you switch. cc @SaxonF @jameskoster